service/service.go

```diff
@@ -82,16 +87,29 @@ func InitService(conf *Config) (svc *Service, err error) {
 
 func initStores(dsn string, cfg *Config) (clientStore *oauth2gorm.ClientStore, tokenStore *oauth2gorm.TokenStore, db *gorm.DB, err error) {
 	//connect database
-	db, err = gorm.Open(postgres.Open(dsn), &gorm.Config{})
-	if err != nil {
-		return nil, nil, nil, err
-	}
+	var sqlDb *sql.DB
+	if cfg.DatadogAgentUrl != "" {
+		sqltrace.Register("pgx", &stdlib.Driver{}, sqltrace.WithServiceName("oauth2server"))
+		sqlDb, err = sqltrace.Open("pgx", dsn)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		db, err = gormtrace.Open(postgres.New(postgres.Config{Conn: sqlDb}), &gorm.Config{})
```

Comment:
Why do you use a different Open argument here than on line 102? It can just be the same, no?

Reply:
Then I wouldn't be using the sqlDb which is set for tracing right?